packages/password-input/src/PasswordInput/PasswordInput.types.ts

```diff
@@ -0,0 +1,117 @@
+import { ChangeEventHandler, ComponentPropsWithoutRef } from 'react';
+
+import { DarkModeProps, Either } from '@leafygreen-ui/lib';
+
+export const States = {
+  Error: 'error', // red x icon, red text
+  Warning: 'warning', // red warning icon, red text
+  Valid: 'valid', // green checkmark, black text
+  None: 'none', // gray checkmark, gray text
+} as const;
+
+export type States = typeof States[keyof typeof States];
+
+export const Size = {
+  Small: 'small',
+  Default: 'default',
+  Large: 'large',
+} as const;
+
+export type Size = typeof Size[keyof typeof Size];
+
+export interface NotificationProps {
+  notification?: string;
+  state: States;
+}
+
+interface AriaLabelProps {
+  /**
+   * Text shown in bold above the input element.
+   *
+   * Optional if `aria-labelledby` or `aria-label` is provided
+   */
+  label?: string;
+
+  /**
+   * Screen-reader label element.
+   *
+   * Optional if `label` or `aria-label` is provided
+   */
+  ['aria-labelledby']?: string;
+
+  /**
+   * Screen reader label text
+   *
+   * Optional if `label` or `aria-labelledby` is provided
+   *
+   */
+  ['aria-label']?: string;
+}
+
+type AriaLabelkeys = keyof AriaLabelProps;
+
+type LabelProps = Either<AriaLabelProps, AriaLabelkeys>;
+
+// Using custom notification container with aria-describedby
+interface StateOnlyProps {
+  /**
+   * The id reference to the custom notification container
+   */
+  ['aria-describedby']: string;
+
+  /**
+   * Determines what notifications will appear
+   *
+   * @default 'none'
+   */
+  stateNotifications: States;
```

Comment:
I alluded to this above, but the api might be more intuitive if we changed the prop name entirely in this case to `state`.

i.e.
```ts
interface StateOnlyProps {
  ['aria-describedby']: string;
  state: States;
  stateNotifications: never;
}
```

Reply:
i initially considered this but dave brought up the point that if we do this then we would have two ways of setting state.  That could become confusing. I've been going back and forth on the name `stateNotifications` since I started but if you have any suggestions i'm open to that.